### ISSUE TITLE
Add flair to post

### ISSUE THREAD

**Issue by Symbian9** [QuantumBadger/RedReader#664]
Add ability to Add/Change post flair

- https://mods.reddithelp.com/hc/en-us/articles/360010513191-Post-Flair

**Comment by Symbian9** [comment]
Any progress on `flair` support? Now many sub-reddits remove/hide posts if flairs not added by post authors.

**Comment by lawson85** [comment]
+1

**Comment by 07416** [comment]
https://github.com/QuantumBadger/RedReader/blob/master/assets/changelog.txt

>50/1.8.6.2  
Post flair support

This "flair support" was only for display?

**Comment by kirk781** [comment]
> Any progress on `flair` support? Now many sub-reddits remove/hide posts if flairs not added by post authors.

Agreed completely. And the numbers of such subreddits is increasing as well. While I find the app to be overall brilliant (especially with the swipe gestures and overall customizable nature) for browsing, it lacks when it comes to posting. Apart from lack of flair support, it also lacks a 'Suggest Title' feature found in many competing apps.

**Comment by Symbian9** [comment]
@QuantumBadger, This issue should be on ***high priority***, because now mostly all sub-reddits, including those which are related to Android (including `/r/FOSSDroid`), just remove/hide posts automatically if newly submitted post has no flair.

**Comment by QuantumBadger** [comment]
Thanks for the info guys. Adding a flair to submitted posts is a feature I'm planning to put in the next release.

Every time the subreddit field in the post submit activity gets modified, we'll have to:

* Request the list of possible flairs from the subreddit
* Check whether a flair is mandatory for that subreddit (hopefully possible using the API) and prevent submission if it's missing
* Show a UI of some sort to allow the flair to be selected

**Comment by Symbian9** [comment]
> Adding a flair to submitted posts is a feature I'm planning to put in the next release.

Thanks, for response. It's really a critical issue for users of such subs as `/r/FOSSDroid`, where posting without flair is prohibited.

- https://old.reddit.com/r/fossdroid/comments/l6z1xv/2_great_foss_reddit_clients_for_android/

**Comment by QuantumBadger** [comment]
@Symbian9 Don't worry, I'm aware :)

I'm very short on free time at the moment, but I've got a prototype of this locally. The UI still needs a fair bit of work before I can push it though.

**Comment by japanesephundroid** [comment]
> @Symbian9 Don't worry, I'm aware :)
> 
> I'm very short on free time at the moment, but I've got a prototype of this locally. The UI still needs a fair bit of work before I can push it though.

Can this be combined with also adding support for showing all award icons since they are probably loaded in a similar way?

**Comment by QuantumBadger** [comment]
I've now committed the initial implementation of this: https://github.com/QuantumBadger/RedReader/commit/3c21f2c0f17863bcdbf87f781231177eb94ca76e

It's available in Alpha 275: https://redreader.org/alpha/

Things not supported yet:

* Enforcing that a flair is provided when it's mandatory
* Adding custom flair text (as opposed to the predefined flairs for the subreddit)

This ended up being a massive change, so if everyone could give the feature plenty of testing in the alpha, that would be much appreciated :) Any problems please let me know!

**Comment by QuantumBadger** [comment]
> Can this be combined with also adding support for showing all award icons since they are probably loaded in a similar way?

@japanesephundroid Unfortunately this uses a different API and would be a separate task. Thanks for the suggestion though :)

**Comment by japanesephundroid** [comment]
> > Can this be combined with also adding support for showing all award icons since they are probably loaded in a similar way?
> 
> @japanesephundroid Unfortunately this uses a different API and would be a separate task. Thanks for the suggestion though :)

Ok, thanks.

**Comment by dekiesel** [comment]
Is the alpha version still the only way to go?
